domains/transfer-agreements/TransferAgreements.API/API.IntegrationTests/Controllers/TransferAgreementsControllerTest.cs

```diff
@@ -44,6 +45,113 @@ public async Task Create_ShouldFail_WhenModelInvalid()
     }
 
+    [Fact]
+    public async Task Create_ShouldFail_WhenStartDateOrEndDateCauseOverlap()
+    {
+        var senderId = Guid.NewGuid();
+        var authenticatedClient = factory.CreateAuthenticatedClient(sub: senderId.ToString());
+        var id = Guid.NewGuid();
+
+        await SeedData(new List<TransferAgreement>()
+        {
+            new()
+            {
+                Id = id,
+                StartDate = DateTimeOffset.UtcNow,
+                EndDate = DateTimeOffset.UtcNow.AddDays(10),
+                ActorId = "actor1",
+                SenderId = senderId,
+                SenderName = "nrgi A/S",
+                SenderTin = "44332211",
+                ReceiverTin = "12345678"
+            }
+        });
+
+        var overlappingRequest = new CreateTransferAgreement(
+            StartDate: DateTimeOffset.UtcNow.AddDays(4).ToUnixTimeSeconds(),
+            EndDate: DateTimeOffset.UtcNow.AddDays(5).ToUnixTimeSeconds(),
+            ReceiverTin: "12345678"
+        );
+
+        var response = await authenticatedClient.PostAsync("api/transfer-agreements", JsonContent.Create(overlappingRequest));
+
+        response.StatusCode.Should().Be(HttpStatusCode.Conflict);
+    }
+
+    [Theory]
+    [InlineData(-1, null, HttpStatusCode.BadRequest, "Start Date cannot be in the past")]
+    [InlineData(-1, 4, HttpStatusCode.BadRequest, "Start Date cannot be in the past")]
+    [InlineData(3, 1, HttpStatusCode.BadRequest, "End Date must be null or later than Start Date")]
+    [InlineData(0, -1, HttpStatusCode.BadRequest, "End Date must be null or later than Start Date")]
+    public async Task Create_ShouldFail_WhenStartOrEndDateInvalid(int startDayOffset, int? endDayOffset, HttpStatusCode expectedStatusCode, string expectedContent)
+    {
+        var senderId = Guid.NewGuid();
+        var authenticatedClient = factory.CreateAuthenticatedClient(sub: senderId.ToString());
+
+        var now = DateTimeOffset.UtcNow;
+
+        var startDate = now.AddDays(startDayOffset).ToUnixTimeSeconds();
+        var endDate = endDayOffset.HasValue ? now.AddDays(endDayOffset.Value).ToUnixTimeSeconds() : (long?)null;
+
+        var request = new CreateTransferAgreement(
+            StartDate: startDate,
+            EndDate: endDate,
+            ReceiverTin: "12345678"
+        );
+
+        var response = await authenticatedClient.PostAsync("api/transfer-agreements", JsonContent.Create(request));
+
+        var validationProblemContent = await response.Content.ReadAsStringAsync();
+
+        response.StatusCode.Should().Be(expectedStatusCode);
+        validationProblemContent.Should().Contain(expectedContent);
+    }
+
+    [Theory]
+    [InlineData(253402300800L, null, "StartDate")]
+    [InlineData(221860025546L, 253402300800L, "EndDate")]
+    public void CreateTransferAgreementEndDateValidator_ShouldValidateMustBeBeforeYear10000(long start, long? end, string? property)
```

Comment:
I think it will make more sense if you put all the test that handle end date validation in one and seperate them from the tests than handle startDate

Reply:
I am not testing per variable basis, but per validation rule. The reason it is separated like this here, is because we only want to test that the format of the long is utctime in seconds and not milliseconds